Update default study def

brianmackenna opened the pull request:
I have updated study definition to separate codelists generation and definition to a codelists.py file. I did different commits so hopefully it is easy to see what I did.

I have also proposed a change to the default dummy cohort size as current is too small for stata playing. 

The test is failing but looks like it is in same pace as @inglesp change so I'll create the PR. Let me know if I need to fix things!

sebbacon commented on .github/workflows/test_runner.yaml:
```diff
@@ -27,7 +27,7 @@ jobs:
         password: ${{ secrets.PACKAGE_READ_ACCESS_TOKEN }}
     - name: Generate dummy data
-      run: cohort-extractor generate_cohort --expectations-population=1000
+      run: cohort-extractor generate_cohort --expectations-population=100000
```
What was the reason for increasing this?

I made it quite small because we're committing it to git and might make the repo very large.  I appreciated 1000 might be too small for testing - but we should pick a sweet spot

brianmackenna replied:
This was from a very brief discussion yesterday about where the stata models converge and what size is needed. 100k was picked for convenience I think.....we didn't discuss if we could hit a sweetspot. @alexwalkercebm ?

alexwalkercebm replied:
Even 10k has not been large enough for some models to converge (e.g. in the ICS repo I had to change it to 100k). It's dependent on the exact model being run, and also because it's randomly generated _sometimes_ 10k was enough, but it needed 100k to be more reliable.

alexwalkercebm replied:
Could we commit a compressed version?

brianmackenna commented:
Thanks for review @sebbacon .

Re didactic comment -> I agree.

- BP -> I removed this without really thinking as I had removed the definition above. I will add back in.
- Heart failure -> I replaced heart disease with this as the definition unlikely to change from study to study. I should add more of the `return_expectations` e.g. incidence or add another clinical codelist like `chronic_liver_disease` back in.
- Smoking -> I actually but this in as it is scary and complex but will also feature in every single study. So for beginners I thought it would mean less scary as they don't have to write it plus saves time for everyone for every study. I had contemplated putting in vaccinations but ruled them out as too scary and not needed for every study.
- Statin -> I should add more of the `return_expectations` commands in. I switched from salbutamol as I think statins will be used in more studies. Unless it is better practice to use specific one offs to make people think a bit more about it?

sebbacon commented:
All your comments sound sensible!




On Fri, 5 Jun 2020, 09:32 Brian, <notifications@github.com> wrote:

> Thanks for review @sebbacon <https://github.com/sebbacon> .
>
> Re didactic comment -> I agree.
>
>    - BP -> I removed this without really thinking as I had removed the
>    definition above. I will add back in.
>    - Heart failure -> I replaced heart disease with this as the
>    definition unlikely to change from study to study. I should add more of the
>    return_expectations e.g. incidence or add another clinical codelist
>    like chronic_liver_disease back in.
>    - Smoking -> I actually but this in as it is scary and complex but
>    will also feature in every single study. So for beginners I thought it
>    would mean less scary as they don't have to write it plus saves time for
>    everyone for every study. I had contemplated putting in vaccinations but
>    ruled them out as too scary and not needed for every study.
>    - Statin -> I should add more of the return_expectations commands in.
>    I switched from salbutamol as I think statins will be used in more studies.
>    Unless it is better practice to use specific one offs to make people think
>    a bit more about it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opensafely/research-template/pull/3#issuecomment-639339017>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABTSRZY7TSAOYGZBIFEQODRVCUS3ANCNFSM4NSY4O7Q>
> .
>

sebbacon commented:
@brianmackenna just checking you plan to finish this? If not let's close!

brianmackenna commented:
Hi - I do plan to finish. I've been meaning to ask with all the changes do I need to close this and start from updated version?

sebbacon commented:
No, should be fine!

CarolineMorton commented:
@brianmackenna I'm tidying up branches. Can I close this?
